backend/wayland/seat.c

```diff
@@ -282,12 +283,40 @@ static struct wlr_wl_input_device *get_wl_input_device_from_input_device(
 	return (struct wlr_wl_input_device *)wlr_dev;
 }
 
+void create_wl_seat(struct wl_seat *wl_seat, struct wlr_wl_backend *wl) {
+	wl->seat.wl_seat = wl_seat;
+	wl_seat_add_listener(wl_seat, &seat_listener, wl);
+}
+
+void destroy_wl_seats(struct wlr_wl_backend *wl) {
```

Comment:
This should cleanup `keyboard` too.

Reply:
This is just re-factoring. I merely moved out exiting code from `backend_destroy`:

```c
	free(wl->seat_name);
	// ...
	if (wl->pointer) {
		wl_pointer_destroy(wl->pointer);
	}
	if (wl->seat) {
		wl_seat_destroy(wl->seat);
	}
```

Note that there is a difference in how pointer and keyboard are handled (see `seat_handle_capabilities`).  
In particular, `wlr_keyboard` created  and destroyed right away as soon as backend is started and exist in single instance. Hence `wl_keyboard_destroy` done in `input_device_destroy` since sole ownership allows it.  
OTOH, `wlr_pointer` is created per output. Thus ownership for host `wl_pointer` will have to be somehow shared and as a simpliest way is to achieve that is lift ownership to object with matching life-cyle e.g. `wlr_wl_backend` originally and `wlr_wl_seat` in this PR (effectively aligned with `wlr_wl_backend`).

Reply:
Hm, OK, I see.